src/lib/consensus/proof_of_stake.ml

```diff
@@ -806,16 +824,175 @@ module Consensus_transition_data = struct
       ~var_of_hlist:of_hlist ~value_to_hlist:to_hlist ~value_of_hlist:of_hlist
 end
 
+module Global_slot = struct
+  open Snark_params.Tick
+
+  include Nat.Make32 ()
```

Comment:
Making this only 32 bits will effectively cap the max number of epochs to `2^16` since at full parameterization the size of the epoch requires 16 bits. This is likely unacceptable.

Reply:
I think it's ok. 2^32 5 second slots is 2^32 * 5 /60/60/24/365 = 680 years.

Reply:
We can/should decrease the # of bits on epochs to 16 bits then.